### PR TITLE
Add tests for redux global state

### PR DESCRIPTION
**Pull request by artpi**
Add tests for https://github.com/Automattic/wp-calypso/pull/2587

### CODE REVIEW

File: client/state/application/actions.js [hunk 1]
```diff
@@ -3,13 +3,12 @@
  */
 import { CONNECTION_LOST, CONNECTION_RESTORED } from 'state/action-types';
 import { warningNotice, successNotice, removeNotice } from 'state/notices/actions';
-import i18n from 'lib/mixins/i18n';
```
Comment (artpi) [inline client/state/application/actions.js]: @rralian : I had to remove it from here because tests were failing on import
Is it a known issue?

Comment (rralian) [inline client/state/application/actions.js]: I think you need to mock it.

Comment (artpi) [inline client/state/application/actions.js]: I moved it out to `network-connection/index.js` but we need a long term solution some day

**Comment by klimeryk** [comment]
LGTM! Hurray for more tests! :tada:

**Pull request merged**